runner/src/runner.rs

```diff
@@ -322,21 +325,39 @@ impl Runner {
         Ok(false)
     }
 
-    fn run_binary(runner: &mut impl RunnerExt, stdout_mode: impl Into<Stdio>) -> Result<Child, Error> {
-        // Ensure there is no other child running
+    fn maybe_run_binary(runner: &mut impl RunnerExt, stdout_mode: impl Into<Stdio>) -> Result<Child, Error> {
```

Comment:
```suggestion
    fn run_binary(runner: &mut impl RunnerExt, stdout_mode: impl Into<Stdio>) -> Result<Child, Error> {
```
I would keep the original naming. To me, the `maybe_` prefix implies that not running running the binary is not unexpected behavior.